server/events/vcs/github_client.go

```diff
@@ -719,9 +719,11 @@ func StatusContextPassed(statusContext StatusContext, vcsstatusname string) bool
 }
 
 func ExpectedCheckPassed(expectedContext githubv4.String, checkRuns []CheckRun, statusContexts []StatusContext, vcsstatusname string) bool {
-	for _, checkRun := range checkRuns {
-		if checkRun.Name == expectedContext {
-			return CheckRunPassed(checkRun)
+	// Iterate through checkRuns from the end, as GitHub returns them in chronological order
```

Comment:
I'm not sure the order can be relied upon. Is it documented that it shall be chronological?

Alternatively, there is the `checkSuite.workflowRun.runNumber` that could be added to the query and used to find the latest check run for the given check.

Reply:
> I'm not sure the order can be relied upon. Is it documented that it shall be chronological?
> 
> Alternatively, there is the `checkSuite.workflowRun.runNumber` that could be added to the query and used to find the latest check run for the given check.

I couldn't find any mentions of order in the documentation; this observation is based solely on empirical evidence.

Thank you for the suggestion. It looks more reliable, and I'll try to implement it 🤝

Reply:
> I'm not sure the order can be relied upon. Is it documented that it shall be chronological?
> 
> Alternatively, there is the `checkSuite.workflowRun.runNumber` that could be added to the query and used to find the latest check run for the given check.

@henriklundstrom  I've rewritten the fix to use `checkSuite.workflowRun.runNumber` for determining the latest check run. Take a look, please.